frame/support/src/traits.rs

```diff
@@ -2148,6 +2148,40 @@ pub trait GetPalletVersion {
 	fn storage_version() -> Option<PalletVersion>;
 }
 
+/// A trait to check the position of inherent in a block.
+///
+/// This is typically implemented on runtime, through `construct_runtime!`.
+pub trait InherentPositionCheck<Block> {
+	/// Check the position of inherents in a block.
+	fn check_inherent_position(block: &Block) -> Result<(), ()>;
+}
+
+/// An extrinsic on which we can get access to call.
+pub trait ExtrinsicCall: sp_runtime::traits::Extrinsic {
```

Comment:
I think this cannot be added inside Extrinsic easily (because OpaqueExtrinsic for instance implements it), thus I extracted it.

Reply:
Looks sensible to me. Maybe I would call it `ExtractCall` or `CallOfExtrinsic`, but looks okay.